<a href="https://colab.research.google.com/github/taryhadani/VINIX7/blob/main/KELOMPOK_2_JULY_DA_VINIX7.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Judul:**

> Analisis Kualitas Pangan dan Pemetaan Dampaknya terhadap Kesehatan Konsumen


---
**SDG:**


*   SDG 2 — Zero Hunger
*   SDG 3 — Good Health and Well-being

> **Dataset:** Open Food Facts Dataset (Kaggle)


# **BAGIAN 1 — Gambaran Dataset**

# **1.1 Import Library**

**Tujuan**

> Mengimpor seluruh library yang diperlukan untuk membaca, mengolah, membersihkan, dan memvisualisasikan dataset.



In [7]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

from google.colab import drive
import os

**Output**


> Tidak menampilkan output apa pun.


---

**Penjelasan**

Library yang digunakan dalam penelitian ini adalah sebagai berikut.

| Library            | Fungsi                                              |
| ------------------ | --------------------------------------------------- |
| pandas             | Membaca dan mengolah dataset dalam bentuk DataFrame |
| numpy              | Melakukan operasi numerik                           |
| matplotlib.pyplot  | Membuat visualisasi data                            |
| seaborn            | Membuat visualisasi statistik                       |
| google.colab.drive | Menghubungkan Google Colab dengan Google Drive      |
| os                 | Memeriksa lokasi dan keberadaan file                |














# **1.2 Menghubungkan Google Drive**

**Tujuan**

> Menghubungkan Google Colab dengan Google Drive agar dataset dapat dibaca.


In [8]:
drive.mount('/content/drive')

Mounted at /content/drive


**Penjelasan**



> Google Drive berhasil dihubungkan dengan Google Colab sehingga file dataset yang berada di dalam Google Drive dapat diakses.



# **1.3 Memastikan Dataset Tersedia**

**Tujuan**

> Memastikan dataset berada pada lokasi penyimpanan yang benar.



In [9]:
path = "/content/drive/MyDrive/en.openfoodfacts.org.products.tsv"

print(os.path.exists(path))

True


**Penjelasan**

Jika hasilnya True, berarti dataset berhasil ditemukan.

Apabila muncul False, berarti lokasi file atau nama file masih salah sehingga perlu diperiksa kembali.

# **1.4 Membaca Dataset**

**Tujuan**

> Membaca dataset Open Food Facts ke dalam DataFrame Pandas



In [10]:
df = pd.read_csv(
    "/content/drive/MyDrive/en.openfoodfacts.org.products.tsv",
    sep="\t",
    low_memory=False
)

**Output**


> Dataset berhasil dimuat ke dalam variabel df.

---

**Penjelasan**


Dataset Open Food Facts menggunakan format **TSV (Tab Separated Values)** sehingga diperlukan parameter **sep="\t"** agar setiap kolom dipisahkan dengan benar. Parameter **low_memory=False** digunakan untuk menghindari peringatan terkait tipe data yang berbeda pada beberapa kolom.


# **1.5 Menampilkan Data Awal**

**Tujuan**

> Menampilkan beberapa data pertama untuk mengetahui struktur awal dataset.



In [11]:
df.head()

,code,url,creator,created_t,created_datetime,last_modified_t,last_modified_datetime,product_name,generic_name,quantity,...,fruits-vegetables-nuts_100g,fruits-vegetables-nuts-estimate_100g,collagen-meat-protein-ratio_100g,cocoa_100g,chlorophyl_100g,carbon-footprint_100g,nutrition-score-fr_100g,nutrition-score-uk_100g,glycemic-index_100g,water-hardness_100g
0,0000000003087,http://world-en.openfoodfacts.org/product/0000...,openfoodfacts-contributors,1474103866,2016-09-17T09:17:46Z,1474103893,2016-09-17T09:18:13Z,Farine de blé noir,NaN,1kg,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,0000000004530,http://world-en.openfoodfacts.org/product/0000...,usda-ndb-import,1489069957,2017-03-09T14:32:37Z,1489069957,2017-03-09T14:32:37Z,Banana Chips Sweetened (Whole),NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,14.0,14.0,NaN,NaN
2,0000000004559,http://world-en.openfoodfacts.org/product/0000...,usda-ndb-import,1489069957,2017-03-09T14:32:37Z,1489069957,2017-03-09T14:32:37Z,Peanuts,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,0.0,0.0,NaN,NaN
3,0000000016087,http://world-en.openfoodfacts.org/product/0000...,usda-ndb-import,1489055731,2017-03-09T10:35:31Z,1489055731,2017-03-09T10:35:31Z,Organic Salted Nut Mix,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,12.0,12.0,NaN,NaN
4,0000000016094,http://world-en.openfoodfacts.org/product/0000...,usda-ndb-import,1489055653,2017-03-09T10:34:13Z,1489055653,2017-03-09T10:34:13Z,Organic Polenta,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


**Penjelasan**



> Perintah **head()** digunakan untuk melihat contoh data yang berhasil dimuat, sehingga peneliti dapat memahami struktur awal dataset sebelum dilakukan analisis lebih lanjut.


# **1.6 Mengetahui Ukuran Dataset**

**Tujuan**

> Mengetahui jumlah baris dan kolom pada dataset.



In [12]:
df.shape

(356027, 163)

**Penjelasan**

Output terdiri dari dua nilai:

*   356.027 baris, yang merepresentasikan data produk makanan.
*   163 kolom, yang berisi berbagai atribut seperti identitas produk, kategori makanan, kandungan nutrisi, informasi kemasan, serta informasi lainnya.


Ukuran dataset ini menunjukkan bahwa Open Food Facts merupakan dataset berskala besar sehingga diperlukan proses persiapan data sebelum dilakukan analisis.

# **1.7 Informasi Dataset**

**Tujuan**

> Mengetahui struktur dataset, tipe data, dan penggunaan memori.



In [14]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 356027 entries, 0 to 356026
Columns: 163 entries, code to water-hardness_100g
dtypes: float64(107), object(56)
memory usage: 442.8+ MB


**Penjelasan**

Berdasarkan hasil **df.info()**, diperoleh informasi sebagai berikut:

| Informasi                     | Hasil         |
| ----------------------------- | ------------- |
| Jumlah data (baris)           | **356.027**   |
| Jumlah atribut (kolom)        | **163**       |
| Tipe data numerik (`float64`) | **107 kolom** |
| Tipe data teks (`object`)     | **56 kolom**  |
| Penggunaan memori             | **442,8 MB**  |



---

Sebagian besar atribut numerik berisi informasi kandungan nutrisi, sedangkan atribut bertipe objek berisi informasi identitas produk, kategori makanan, merek, negara asal, dan deskripsi lainnya.

# **1.8 Statistik Deskriptif Dataset**

**Tujuan**

> Mengetahui karakteristik awal data numerik melalui statistik deskriptif, seperti jumlah data, nilai rata-rata, standar deviasi, nilai minimum, nilai maksimum, dan kuartil.


In [16]:
df.describe()

,no_nutriments,additives_n,ingredients_from_palm_oil_n,ingredients_from_palm_oil,ingredients_that_may_be_from_palm_oil_n,ingredients_that_may_be_from_palm_oil,nutrition_grade_uk,energy_100g,energy-from-fat_100g,fat_100g,...,fruits-vegetables-nuts_100g,fruits-vegetables-nuts-estimate_100g,collagen-meat-protein-ratio_100g,cocoa_100g,chlorophyl_100g,carbon-footprint_100g,nutrition-score-fr_100g,nutrition-score-uk_100g,glycemic-index_100g,water-hardness_100g
count,0.0,283867.000000,283867.000000,0.0,283867.000000,0.0,0.0,295367.000000,869.000000,2.794970e+05,...,3228.000000,404.000000,182.000000,1383.000000,0.0,278.000000,254856.000000,254856.000000,0.0,0.0
mean,NaN,1.876851,0.023430,NaN,0.059736,NaN,NaN,1125.453320,587.216617,5.606587e+04,...,33.392680,60.360124,15.362637,52.102675,NaN,335.790664,9.166137,8.980656,NaN,NaN
std,NaN,2.501022,0.153094,NaN,0.280660,NaN,NaN,936.825952,713.255708,2.963385e+07,...,32.906834,29.262350,3.692658,19.028361,NaN,423.244817,8.999870,9.151757,NaN,NaN
min,NaN,0.000000,0.000000,NaN,0.000000,NaN,NaN,0.000000,0.000000,0.000000e+00,...,0.000000,0.000000,8.000000,6.000000,NaN,0.000000,-15.000000,-15.000000,NaN,NaN
25%,NaN,0.000000,0.000000,NaN,0.000000,NaN,NaN,382.000000,49.400000,1.000000e-01,...,0.000000,45.000000,12.000000,33.000000,NaN,82.650000,1.000000,1.000000,NaN,NaN
50%,NaN,1.000000,0.000000,NaN,0.000000,NaN,NaN,1092.000000,300.000000,5.290000e+00,...,25.000000,58.000000,15.000000,52.000000,NaN,190.950000,10.000000,9.000000,NaN,NaN
75%,NaN,3.000000,0.000000,NaN,0.000000,NaN,NaN,1674.000000,900.000000,2.000000e+01,...,55.000000,93.000000,15.000000,70.000000,NaN,378.700000,16.000000,16.000000,NaN,NaN
max,NaN,30.000000,2.000000,NaN,6.000000,NaN,NaN,231199.000000,3830.000000,1.566667e+10,...,100.000000,100.000000,25.000000,100.000000,NaN,2842.000000,40.000000,40.000000,NaN,NaN


**Penjelasan**

> Berdasarkan hasil df.describe(), dataset memiliki 107 atribut numerik dengan jumlah data yang berbeda pada setiap kolom. Hal ini menunjukkan masih terdapat missing value pada beberapa atribut. Selain itu, terdapat beberapa nilai maksimum yang sangat tinggi sehingga mengindikasikan kemungkinan adanya outlier. Informasi ini menjadi dasar dalam proses pembersihan data sebelum dilakukan analisis lebih lanjut.



# **1.9 Deskripsi Dataset**

> Dataset yang digunakan dalam penelitian ini adalah Open Food Facts yang diperoleh dari platform Kaggle. Dataset ini berisi informasi mengenai berbagai produk makanan dari berbagai negara di dunia, termasuk nama produk, merek, kategori makanan, negara asal, kandungan nutrisi, jumlah zat aditif, serta penilaian kualitas nutrisi (nutrition grade). Informasi tersebut dapat dimanfaatkan untuk menganalisis kualitas pangan berdasarkan kandungan nutrisi yang dimiliki setiap produk.



# **1.10 Sumber Dataset**


*   **Nama Dataset:** Open Food Fact
*  **Sumber:** https://www.kaggle.com/datasets/openfoodfacts/world-food-facts
*   **Format Data:** TSV (Tab Separated Values)






# **1.11 Keterkaitan Dataset dengan Permasalahan**

> Penelitian ini mengangkat topik **Analisis Kualitas Pangan dan Pemetaan Dampaknya terhadap Kesehatan Konsumen** yang mendukung **SDG 2 (Zero Hunger) dan SDG 3 (Good Health and Well-being)**. Dataset Open Food Facts dipilih karena menyediakan informasi mengenai kandungan energi, lemak, lemak jenuh, gula, serat, protein, garam, serta *nutrition grade* pada berbagai produk makanan. Informasi tersebut memungkinkan analisis terhadap kualitas nutrisi produk dan identifikasi pola pangan yang berpotensi memengaruhi kesehatan konsumen. Hasil analisis diharapkan dapat memberikan gambaran mengenai kualitas pangan yang beredar serta meningkatkan kesadaran masyarakat dalam memilih produk makanan yang lebih sehat.



# **BAGIAN 2 - Pembersihan & Persiapan Data**

## **2.1 Seleksi Variabel Penelitian**

**Tujuan**

> Memilih atribut yang relevan dengan tujuan penelitian agar proses analisis lebih terfokus.



In [17]:
kolom = [
    'product_name',
    'brands',
    'categories',
    'countries',
    'additives_n',
    'nutrition_grade_fr',
    'energy_100g',
    'fat_100g',
    'saturated-fat_100g',
    'sugars_100g',
    'fiber_100g',
    'proteins_100g',
    'salt_100g'
]

df = df[kolom]

| No | Variabel             |
| -- | -------------------- |
| 1  | `product_name`       |
| 2  | `brands`             |
| 3  | `categories`         |
| 4  | `countries`          |
| 5  | `nutrition_grade_fr` |
| 6  | `energy_100g`        |
| 7  | `fat_100g`           |
| 8  | `saturated-fat_100g` |
| 9  | `sugars_100g`        |
| 10 | `fiber_100g`         |
| 11 | `proteins_100g`      |
| 12 | `salt_100g`          |
| 13 | `additives_n`        |


**Output**

> Dataset berubah dari 163 kolom menjadi 13 kolom yang sesuai dengan penelitian.


---


**Penjelasan**

> Variabel dipilih karena berkaitan dengan identitas produk, kualitas nutrisi, dan kandungan gizi yang akan dianalisis.



## **1.2 Menampilkan Data Setelah Seleksi**

**Tujuan**

> Memastikan hanya variabel yang dipilih yang tersisa pada dataset.


In [18]:
df.head()

,product_name,brands,categories,countries,additives_n,nutrition_grade_fr,energy_100g,fat_100g,saturated-fat_100g,sugars_100g,fiber_100g,proteins_100g,salt_100g
0,Farine de blé noir,Ferme t'y R'nao,NaN,en:FR,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,Banana Chips Sweetened (Whole),NaN,NaN,US,0.0,d,2243.0,28.57,28.57,14.29,3.6,3.57,0.00000
2,Peanuts,Torn & Glasser,NaN,US,0.0,b,1941.0,17.86,0.00,17.86,7.1,17.86,0.63500
3,Organic Salted Nut Mix,Grizzlies,NaN,US,0.0,d,2540.0,57.14,5.36,3.57,7.1,17.86,1.22428
4,Organic Polenta,Bob's Red Mill,NaN,US,0.0,NaN,1552.0,1.43,NaN,NaN,5.7,8.57,NaN


**Output**



> Menampilkan lima baris pertama dari dataset hasil seleksi.




---


**Penjelasan**


> Tahap ini dilakukan untuk memastikan proses seleksi variabel telah berhasil.



## **1.3 Mengecek Ukuran Dataset**

**Tujuan**

> Mengetahui jumlah baris dan kolom setelah seleksi variabel.



In [19]:
df.shape

(356027, 13)

**Output**

> Menampilkan jumlah baris dan 13 kolom hasil seleksi.


---


**Penjelasan**

> Output menunjukkan ukuran dataset yang akan digunakan pada proses pembersihan data.



# **1.4 Mengecek Missing Value**

**Tujuan**


> Mengetahui jumlah data kosong pada setiap atribut.



In [20]:
df.isnull().sum()

,0
product_name,17512
brands,29050
categories,252728
countries,275
additives_n,72160
nutrition_grade_fr,101171
energy_100g,60660
fat_100g,76530
saturated-fat_100g,92204
sugars_100g,76841


**Output**

> Menampilkan jumlah missing value pada setiap kolom.


---


Penjelasan
**teks tebal**
> Hasil ini digunakan untuk menentukan strategi penanganan data kosong pada tahap berikutnya.




# **1.5 Mengecek Data Duplikat**

**Tujuan**


> Mengetahui apakah terdapat data yang sama atau berulang.



In [21]:
df.duplicated().sum()

np.int64(19743)

**Output**

> Menampilkan jumlah data duplikat.

---


**Penjelasan**

> Data duplikat dapat memengaruhi hasil analisis sehingga perlu diperiksa sebelum dihapus.

# **1.7 Menghapus Data Duplikat**

In [29]:
df = df.drop_duplicates()



> lalu dicek kembali menggunakan:



In [30]:
df.duplicated().sum()

np.int64(0)



> Ini menunjukkan bahwa seluruh data duplikat telah berhasil dihapus.



# **1.8 Mengecek Tipe Data**

**Tujuan**

> Memastikan tipe data setiap atribut sudah sesuai.



In [32]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 331441 entries, 0 to 356026
Data columns (total 13 columns):
 #   Column              Non-Null Count   Dtype  
---  ------              --------------   -----  
 0   product_name        331441 non-null  object 
 1   brands              331441 non-null  object 
 2   categories          331441 non-null  object 
 3   countries           331441 non-null  object 
 4   additives_n         331441 non-null  float64
 5   nutrition_grade_fr  331441 non-null  object 
 6   energy_100g         331441 non-null  float64
 7   fat_100g            331441 non-null  float64
 8   saturated-fat_100g  331441 non-null  float64
 9   sugars_100g         331441 non-null  float64
 10  fiber_100g          331441 non-null  float64
 11  proteins_100g       331441 non-null  float64
 12  salt_100g           331441 non-null  float64
dtypes: float64(8), object(5)
memory usage: 35.4+ MB


**Output**

> Menampilkan tipe data setiap kolom.

---

**Penjelasan**


> Tahap ini memastikan atribut numerik dapat digunakan untuk analisis statistik.



# **1.9 Menangani Missing Value**

**Tujuan**

> Menangani nilai kosong pada dataset agar seluruh variabel siap digunakan dalam proses analisis.



In [33]:
# Menghapus data yang tidak memiliki nama produk
df = df.dropna(subset=['product_name'])

# Mengisi nilai kosong pada data kategorikal
df['brands'] = df['brands'].fillna('Unknown')
df['categories'] = df['categories'].fillna('Unknown')
df['countries'] = df['countries'].fillna('Unknown')

# Mengisi nilai kosong pada data numerik dengan nilai median
kolom_numerik = [
    'additives_n',
    'energy_100g',
    'fat_100g',
    'saturated-fat_100g',
    'sugars_100g',
    'fiber_100g',
    'proteins_100g',
    'salt_100g'
]

for kolom in kolom_numerik:
    df[kolom] = df[kolom].fillna(df[kolom].median())

# Mengisi nilai kosong pada nutrition grade
df['nutrition_grade_fr'] = df['nutrition_grade_fr'].fillna('Unknown')

In [36]:
df.isnull().sum()

,0
product_name,0
brands,0
categories,0
countries,0
additives_n,0
nutrition_grade_fr,0
energy_100g,0
fat_100g,0
saturated-fat_100g,0
sugars_100g,0


**Penjelasan**

> Proses penanganan *missing value* dilakukan dengan menghapus data yang tidak memiliki nama produk, mengisi nilai kosong pada variabel kategorikal menggunakan nilai **"Unknown"**, serta mengisi nilai kosong pada variabel numerik menggunakan median. Metode median dipilih karena lebih stabil terhadap nilai ekstrem ***(outlier)***. Setelah proses tersebut dilakukan, seluruh variabel tidak lagi memiliki *missing value*, sehingga dataset siap digunakan untuk tahap analisis selanjutnya.



# **1.10 Memastikan Dataset Bersih**

**Tujuan**

> Memastikan bahwa proses penanganan *missing value* telah berhasil dilakukan sehingga tidak terdapat lagi nilai kosong pada dataset.



In [37]:
df.isnull().sum()

,0
product_name,0
brands,0
categories,0
countries,0
additives_n,0
nutrition_grade_fr,0
energy_100g,0
fat_100g,0
saturated-fat_100g,0
sugars_100g,0


**Output**

> Seluruh kolom memiliki nilai 0, yang menunjukkan bahwa tidak ada lagi missing value pada dataset.

---


**Penjelasan**

> Berdasarkan hasil pengecekan, seluruh variabel memiliki nilai 0 pada missing value. Hal ini menunjukkan bahwa proses pembersihan data telah berhasil dilakukan dan tidak terdapat lagi data kosong pada dataset. Dengan demikian, dataset telah memiliki kualitas data yang lebih baik dan siap digunakan pada tahap eksplorasi data ***(Exploratory Data Analysis/EDA)*** dan analisis selanjutnya.





# **1.11 Dataset Siap Digunakan**

**Penjelasan**

> Setelah dilakukan proses seleksi variabel, penghapusan data duplikat, pengecekan tipe data, dan penanganan missing value, dataset telah memiliki kualitas data yang lebih baik. Dataset ini selanjutnya dapat digunakan untuk proses eksplorasi data dan analisis guna memperoleh informasi mengenai kualitas pangan dan dampaknya terhadap kesehatan konsumen.



In [38]:
import pandas as pd

ringkasan = pd.DataFrame({
    "Keterangan": [
        "Jumlah data sebelum cleaning",
        "Jumlah data setelah cleaning",
        "Jumlah data yang dihapus"
    ],
    "Jumlah": [
        356027,
        len(df),
        356027 - len(df)
    ]
})

ringkasan

,Keterangan,Jumlah
0,Jumlah data sebelum cleaning,356027
1,Jumlah data setelah cleaning,331441
2,Jumlah data yang dihapus,24586
